app/src/org/commcare/android/database/global/models/ApplicationRecord.java

```diff
@@ -56,4 +82,102 @@ public int getStatus() {
     public void setStatus(int status) {
         this.status = status;
     }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public void setUniqueId(String id) {
+        this.uniqueId = id;
+    }
+
+    public String getDisplayName() {
+        return this.displayName;
+    }
+
+    public void setDisplayName(String appName) {
+        this.displayName = appName;
+    }
+
+    public int getVersionNumber() {
+        return this.versionNumber;
+    }
+
+    public void setVersionNumber(int version) {
+        this.versionNumber = version;
+    }
+
+    public void setArchiveStatus(boolean b) {
+        this.isArchived = b;
+    }
+
+    public boolean isArchived() {
+        return this.isArchived;
+    }
+
+    public void setResourcesStatus(boolean b) {
+        this.resourcesValidated = b;
+    }
+
+    public boolean resourcesValidated() {
+        return this.resourcesValidated;
+    }
+
+    // A 'visible' app record has status installed and is not archived
+    public boolean isVisible() {
+        return status == STATUS_INSTALLED && !isArchived;
+    }
+
+    // A 'usable' app record is one that will actually get shown to a user -- is visible and has
+    // its MM resources validated
+    public boolean isUsable() {
+        return isVisible() && resourcesValidated;
+    }
+
+    @Override
+    public String toString() {
+        return this.displayName;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord represents an app generated from
+     * a pre-Multiple Apps version of CommCare that does not have profile files
+     * with uniqueId and displayName
+     */
+    public boolean preMultipleAppsProfile() {
+        return this.preMultipleAppsProfile;
+    }
+    
+    public void setPreMultipleAppsProfile(boolean b) {
+        this.preMultipleAppsProfile = b;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord was just generated from the a
+     * different ApplicationRecord format via the db upgrader, because it was 
+     * initially installed on a phone with a pre-Multiple Apps version of CommCare
+     */
+    public boolean convertedByDbUpgrader() {
+        return this.convertedViaDbUpgrader;
+    }
+
+    public void setConvertedByDbUpgrader(boolean b) {
+        this.convertedViaDbUpgrader = b;
+    }
+
+    /**
+     * Used when this record is first installed or upgraded from an old version, to set all
+     * properties of the record that come from its profile file
+     */
+    public void setPropertiesFromProfile(Profile p) {

```

Comment:
Feel free to ignore: I think it would be helpful to collect all of the methods here that pertain only to the upgrade process and make sure they are visually/clearly distinguished so people don't think they are relevant for them.